yarn-project/noir-libs/noir-aztec/src/note/note_getter.nr

```diff
@@ -27,67 +27,65 @@ fn check_note_header<Note, N>(
 }
 
 fn ensure_note_exists<Note, N>(
-    mut context: Context,
+    context: &mut Context,
     storage_slot: Field,
     note_interface: NoteInterface<Note, N>,
-    note: Note,
-) -> (Context, Note) {
-    let mut note_with_header = note;
+    note: &mut Note,
+) {
     let mut unique_note_hash = 0;
     let is_dummy = note_interface.is_dummy;
-    if is_dummy(note) == false {
+    if is_dummy(*note) == false {
         // Get a note from oracle and early out if it doesn't exist.
         let saved_note = get_note_internal(storage_slot, note_interface);
         assert(is_dummy(saved_note) == false);
 
         // Only copy over the header to the original note to make sure the preimage is the same.
         let get_header = note_interface.get_header;
         let set_header = note_interface.set_header;
-        let header = get_header(saved_note);
-        note_with_header = set_header(note, header);
+        let note_header = get_header(saved_note);
+        set_header(note, note_header);
 
-        check_note_header(context, storage_slot, note_interface, note_with_header);
+        check_note_header(*context, storage_slot, note_interface, *note);
 
-        unique_note_hash = compute_unique_note_hash(note_interface, note_with_header);
+        unique_note_hash = compute_unique_note_hash(note_interface, *note);
     };
-    context = context.push_read_request(unique_note_hash);
-    (context, note_with_header)
+    context.push_read_request(unique_note_hash);
 }
 
 fn get_note<Note, N>(
-    mut context: Context,
+    context: &mut Context,
     storage_slot: Field,
     note_interface: NoteInterface<Note, N>,
-) -> (Context, Note) {
+) -> Note {
     let note = get_note_internal(storage_slot, note_interface);
     let mut unique_note_hash = 0;
     let is_dummy = note_interface.is_dummy;
     if is_dummy(note) == false {
-        check_note_header(context, storage_slot, note_interface, note);
+        check_note_header(*context, storage_slot, note_interface, note);
```

Comment:
Hold on, why `*`? Do we need to manually dereference the mutable reference?

Reply:
Yeah, since `check_note_header` doesn't mutate the context, we want to pass in a `Context` type rather than an `&mut Context` type